### PR TITLE
feat(compute): add terraform sample for instanceflexibilitypolicy of rmig

### PR DESCRIPTION
**Pull request by chayankumar999**
## Description

Fixes #333070003

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [x] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [x] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [x] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [x] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [x] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [x] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s):

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [ ] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ ] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

File: compute/rmig_instance_flexibility_policy/main.tf [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      machine_types = [ "n2-standard-1" ]
+    }
+    instance_selections {
+      name = "if-nothing-else"
+      rank = 3
+      machine_types = [ "e2-standard-2" ]
+    }
+  }
+
+  update_policy {
```
Comment (askubis) [inline compute/rmig_instance_flexibility_policy/main.tf:69]: is this mandatory for the command to work?

Comment (askubis) [reply compute/rmig_instance_flexibility_policy/main.tf:69]: I eman everything but the instance_redistribution policy

Comment (chayankumar999) [reply compute/rmig_instance_flexibility_policy/main.tf:69]: I think yes. Last time when run the test it failed with minimal_action validation error.

**Comment by glasnt** [comment]
FYI: I know this PR isn't ready for review yet, but you're getting this error: 

> The argument "base_instance_name" is required, but no definition was found.

File: compute/rmig_instance_flexibility_policy/main.tf [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ /**
+ * Made to resemble:
+ * gcloud alpha compute instance-groups managed create flex-igm --project=$PROJECT --region=us-central1 \
+ * --target-distribution-shape=any-single-zone --instance-redistribution-type NONE \
+ * --template $TEMPLATE --size 3 \
```
[the remark this answers is not in the document]
Comment (chayankumar999) [reply compute/rmig_instance_flexibility_policy/main.tf]: Done

**Comment by chayankumar999** [comment]
> FYI: I know this PR isn't ready for review yet, but you're getting this error:
> 
> > The argument "base_instance_name" is required, but no definition was found.

Done!

**Review by askubis — APPROVED** [review]

**Comment by glasnt** [comment]
The tests in this PR pass only when I use the 6.12 google-beta provider; is this intended? 

(The use of this version is causing crashes elsewhere)

**Comment by chayankumar999** [comment]
> The tests in this PR pass only when I use the 6.12 google-beta provider; is this intended?
> 
> (The use of this version is causing crashes elsewhere)

The terraform feature is still not available GA.  As I can see, you added the provider along with version.  That means when the GA will be available, I have to submit another PR with mentioning the google provider. Is it so?

**Review by glasnt — APPROVED** [review]

**Comment by glasnt** [comment]
When this functionality is in the main google provider (not just `google-beta`), there will need to be a change to (currently) line 31 to specify `google` rather than `google-beta`. 

This information is outside the region tags, so the contents of the sample will remain the same pre and post GA

**Comment by chayankumar999** [comment]
> tags

Got it.
Thanks :)

**Pull request merged**